### PR TITLE
Cherrypick Eiki's Task Edit Suggestions Part 2

### PR DESCRIPTION
**Pull request by EvianTan**
This PR is the second part of [Eiki's Task Edit Suggestions PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/497/commits), due to the large scale of changes and commits, we decided to cherrypick all his changes and split into couple new PRs, which would be more readable and understandable. You can also see the first cherrypick PR [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/536)

**Mainly changes explained:** 🔮 
1. delete all the mock data on the frontend, change to populate data from backend
2. create the ability to do tasks suggestions for volunteer/core team user
3. implement the reject function on the taskEditSuggestion

**How to test:** 🔍 

1. check on the [related backend PR](https://github.com/OneCommunityGlobal/HGNRest/pull/186)
2. check on this PR as frontend
4. log as volunteer/core team account
5. Dashboard -> Tasks and Timelogs -> Tasks tab -> click one of the tasks -> click the "suggest" button -> make some changes on the popup table -> **click "update" button**
image

6. log out and log in as a admin/owner user
7. go to http://localhost:3000/taskeditsuggestions
8. click on the row on taskEditSuggestions page to open the suggested changes, click the "Reject" button
image

**Note: below functionalities will be implemented in coming PRs** 💡 
1. tasks suggest function only be accessed by manager/mentor users, instead, volunteer/core team users can only view tasks
2. taskEditSuggestion page will only be accessed by admin/owner users

### CODE REVIEW

**Comment by guiwustro** [comment]
The feature is all ok! The only thing is that has 2 class CSS named as "class" instead of "className" that was preventing to render the /taskeditsuggestions in my localhost. It's not in your change files, but in the branch It's in this way:
Component: TaskEditSuggestions.jsx
https://prnt.sc/xnX_fDSp1yPw

And also, as you already say, adding the hasPermission function to block the route for no owners/administrator.
I loved your PR, very detailed and easy to test.

**Review by guiwustro — APPROVED** [review]
The feature is all ok! The only thing is that has 2 class CSS named as "class" instead of "className" that was preventing to render the /taskeditsuggestions in my localhost. It's not in your change files, but in the branch It's in this way:
Component: TaskEditSuggestions.jsx
https://prnt.sc/xnX_fDSp1yPw

And also, as you already say, adding the hasPermission function to block the route for no owners/administrator.
I loved your PR, very detailed and easy to test.

**Comment by EvianTan** [comment]
> The feature is all ok! The only thing is that has 2 class CSS named as "class" instead of "className" that was preventing to render the /taskeditsuggestions in my localhost. It's not in your change files, but in the branch It's in this way: Component: TaskEditSuggestions.jsx https://prnt.sc/xnX_fDSp1yPw
> 
> And also, as you already say, adding the hasPermission function to block the route for no owners/administrator. I loved your PR, very detailed and easy to test.

Great catch! I'll check that later.

**Comment by EvianTan** [comment]
> The feature is all ok! The only thing is that has 2 class CSS named as "class" instead of "className" that was preventing to render the /taskeditsuggestions in my localhost. It's not in your change files, but in the branch It's in this way: Component: TaskEditSuggestions.jsx https://prnt.sc/xnX_fDSp1yPw
> 
> And also, as you already say, adding the hasPermission function to block the route for no owners/administrator. I loved your PR, very detailed and easy to test.

Hi @guiwustro ,

I just checked into the backend branch and this frontend branch, and the url http://localhost:3000/taskeditsuggestions rendered well. And I also checked the two class CSS you mentioned above, after toggling the arrow button, they also worked well, please see below images. So I don't think that's the reason you cannot render this page. **Were you on the correct branches?**
image
image


I didn't pay attention to the `class` instead of `className` before you pointed out, then I did some research and found `className` is recommended than `class` in modern react. But if you insist to use `class`, you will get just a warning but nothing affects on output. [(Reference link)](https://www.geeksforgeeks.org/why-react-uses-classname-over-class-attribute/)
image

File: src/components/Projects/WBS/WBSDetail/EditTask/EditTaskModal.jsx [hunk 1]
```diff
@@ -253,7 +253,7 @@ const EditTaskModal = props => {
       classification,
     };
 
-    props.updateTask(props.taskId, updatedTask);
+    props.updateTask(props.taskId, updatedTask, hasPermission(role, 'editTask'));
```
Comment (guiwustro) [inline src/components/Projects/WBS/WBSDetail/EditTask/EditTaskModal.jsx:256]: hasPermission should receive 4 parameters (role, action, roles, userPermissions). The roles is the state that get all roles and permissions of each one in the API, and the userPermissions take the permissions of an individual user (if he had)
check line 630 (for example)

File: src/components/TaskEditSuggestions/TaskEditSuggestions.jsx [hunk 2]
```diff
@@ -1,252 +1,24 @@
...
   const [isTaskEditSuggestionModalOpen, setIsTaskEditSuggestionModalOpen] = useState(false);
   const [currentTaskEditSuggestion, setCurrentTaskEditSuggestion] = useState();
-  
-  const [userSortDirection, setUserSortDirection] = useState();
-  const [dateSuggestedSortDirection, setDateSuggestedSortDirection] = useState("desc");
 
-  useEffect(() => setTaskEditSuggestions(taskEditSuggestionsInitial.sort((tes1, tes2) => tes2.dateSuggested.localeCompare(tes1.dateSuggested))), []);
+  const { isLoading, taskEditSuggestions, userSortDirection, dateSuggestedSortDirection, userRole } = useSelector(getTaskEditSuggestionsData);
+
+  const isUserPermitted = hasPermission(userRole, 'editTask');
```
Comment (guiwustro) [inline src/components/TaskEditSuggestions/TaskEditSuggestions.jsx:18]: same that the previous comment. has permissions needs 4 parameters

**Pull request merged**